src/components/editMeasure/details/EditMeasureDetailsSideNav.tsx

```diff
@@ -18,10 +24,14 @@ export default function EditMeasureDetailsSideNav(
 ) {
   const { links } = props;
   const { pathname } = useLocation();
-  const history = useHistory();
-
+  const endRoute = /[^/]*$/.exec(pathname)[0];
+  const { id } = useParams<{
+    id: string;
+  }>();
+  let navigate = useNavigate();
   const handleChange = (e, v) => {
-    history.push(v);
+    const newPath = `/measures/${id}/edit/details/${v}`;
+    navigate(newPath, { replace: true });
```

Comment:
Do we need to replace the history ? I am not sure what is the down side of letting users navigate back to previous route using browser history 

`If using replace: true, the navigation will replace the current entry in the history stack instead of adding a new one.`

Reply:
No we don't want that. Another learning artifact. This functionally ruins the capacity for a backbutton.